src/kixi/hecuba/onyx/jobs/weather_fetcher.clj

```diff
@@ -35,31 +60,57 @@
 
 (defn pull-data
   "Get Metoffice data as a string"
-  ([querydate querytime siteid]
-   (->> (client/post "http://datagovuk.cloudapp.net/query"
-                     {:form-params {:Type "Observation"
-                                    :PredictionSiteID siteid
-                                    :ObservationSiteID siteid
-                                    :Date querydate ;; dd/mm/yyyy
-                                    :PredictionTime querytime ;; 0000
-                                    }
-                      :follow-redirects true})
-        :body
-        (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
-        slurp
-        )))
+  [querydate querytime siteid]
+  (->> (client/post "http://datagovuk.cloudapp.net/query"
+                    {:form-params {:Type "Observation"
+                                   :PredictionSiteID siteid
+                                   :ObservationSiteID siteid
+                                   :Date querydate ;; dd/mm/yyyy
+                                   :PredictionTime querytime ;; 0000
+                                   }
+                     :follow-redirects true})
+       :body
+       (re-find #"https://datagovuk.blob.core.windows.net/csv/[a-z0-9]+.csv")
+       (client/get)
+       :body
+       ))
 
 (defn pull-weather-station-day-data [querydate siteid]
   (mapv (fn [hour] (try (-> (pull-data querydate (format "%02d00" hour) siteid)
                             (process-data-str)
                             (keep-temp-date-time))
-                        (catch Exception e (str "Exception caught: " (.getMessage e)))))
+                        (catch Exception e (str "Exception caught: " (.getMessage e)))
+                        ))
```

Comment:
tidy parens, and what will you do with the nils that you might get here? Should you be doing `(into [] (keep ...) (range 24))` instead?